### PR TITLE
Rename zNext to z14

### PR DESCRIPTION
**Pull request by fjeremic**
z14 has been announced on July 17, 2017. In light of this rename zNext
to z14. In addition leave all zNext definitions in tact for use in
future iterations of the hardware.

Signed-off-by: Filip Jeremic <fjeremic@ca.ibm.com>

### CODE REVIEW

File: compiler/z/codegen/OMRCodeGenerator.cpp [hunk 1]
```diff
@@ -457,6 +457,15 @@ TR_S390ProcessorInfo::checkZ13()
    }
 
 ////////////////////////////////////////////////////////////////////////////////
+// TR_S90ProcessorInfo::checkZ14: returns true if it's z14, otherwise false.
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: Should be:  TR_S390ProcessorInfo::

File: compiler/z/codegen/OMRCodeGenerator.cpp [hunk 2]
```diff
@@ -457,6 +457,15 @@ TR_S390ProcessorInfo::checkZ13()
...
+////////////////////////////////////////////////////////////////////////////////
+bool
+TR_S390ProcessorInfo::checkZ14()
+   {
+   return TR::Compiler->target.cpu.getS390SupportsZ14();
+   }
+
+////////////////////////////////////////////////////////////////////////////////
 // TR_S90ProcessorInfo::checkZNext: returns true if it's zNext, otherwise false.
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: Should be:  TR_S390ProcessorInfo::

Comment (fjeremic) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: Fixed in cb358ed. I also fixed a few other occurrences while in this area of code.

File: compiler/z/codegen/OMRCodeGenerator.cpp (hunk 1 above)
Comment (fjeremic) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: Fixed in cb358ed. I also fixed a few other occurrences while in this area of code.

**Pull request merged**